src/public/app/widgets/tab_row.js

```diff
@@ -267,8 +267,28 @@ export default class TabRowWidget extends BasicWidget {
                 }
             });
         });
+
+        keyboardActionService.setupActionsForElement('tabs', $(document), this);
+    }
+
+    goToTab(tabNumber) {
+        const index = tabNumber === 0 ? this.tabEls.length - 1 : tabNumber - 1;
+        const tab = this.tabEls[index];
+        if (!tab) return;
+        appContext.tabManager.activateNoteContext(tab.getAttribute('data-ntx-id'));
     }
 
+    firstTabCommand() {this.goToTab(1);}
+    secondTabCommand() {this.goToTab(2);}
+    thirdTabCommand() {this.goToTab(3);}
+    fourthTabCommand() {this.goToTab(4);}
+    fifthTabCommand() {this.goToTab(5);}
+    sixthTabCommand() {this.goToTab(6);}
+    seventhTabCommand() {this.goToTab(7);}
+    eigthTabCommand() {this.goToTab(8);}
+    ninthTabCommand() {this.goToTab(9);}
+    lastTabCommand() {this.goToTab(0);}
```

Comment:
Instead of 0, I'd use something more distinct like -1 or 999. 0 is often used for the first element and thus could be ambiguous.

Reply:
I think `-1` could get confused but index not found, but I will definintely use something more unique than `0`.